src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -364,6 +364,7 @@ protected void syncWithLeader(long newLeaderZxid) throws Exception{
         readPacket(qp);
         LinkedList<Long> packetsCommitted = new LinkedList<Long>();
         LinkedList<PacketInFlight> packetsNotCommitted = new LinkedList<PacketInFlight>();
+        boolean syncSnapshot = false;
```

Comment:
We can level this variable definition up so it's clustered with `snapshotNeed` boolean.

Another possibility is to get ride of this variable and use existing `snapshotNeeded` - but that will do fysnc snapshot for TRUNC sync, which the existing patch will not.